tasks/main.yml

```
@@ -10,6 +10,22 @@
   set_fact:
     ansible_python_interpreter: "{{ netbox_python_binary }}"
 
+- name: Recheck new Python interpreter's version
+  setup:
+    gather_subset: min
+    filter: ansible_python_version
+
+# Can't quite check the effective version of a git installation this early since
+# the repo isn't checked out, but most users will be on develop in the first
+# place so this dumb check should be good enough.
+- name: Ensure the installed Python version is newer than 3.6.0 for NetBox 2.8+
+  assert:
+    that:
+      - "ansible_python_version | version_compare('3.6.0', '>=')"
+    msg: "NetBox 2.8.0+ requires Python 3.6+. Please either specify an older NetBox version or upgrade to a newer distribution that provides Python 3.6+."
+  when:
+    - (netbox_stable and netbox_stable_version | version_compare('2.8.0', '>=') or netbox_git
+
```

This does not work for me on 2.9.7 but the below does:
```suggestion
- name: Ensure the installed Python version is newer than 3.6.0 for NetBox 2.8+
  assert:
    that:
      - "ansible_python_version is version('3.6.0', '>=')"
    msg: "NetBox 2.8.0+ requires Python 3.6+. Please either specify an older NetBox version or upgrade to a newer distribution that provides Python 3.6+."
  when:
    - netbox_stable and netbox_stable_version is version('2.8.0', '>=') or netbox_git
```

Not sure if this is a discrepancy due to ansible versions and the versions test being renamed?

---

https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#version-comparison

Yeah, looks like `version_compare` was renamed in 2.5 and I didn't notice.